Source/Core/Absy.cs

```diff
@@ -1457,6 +1458,36 @@ public DatatypeTypeCtorDecl(TypeCtorDecl typeCtorDecl)
       : base(typeCtorDecl.tok, typeCtorDecl.Name, typeCtorDecl.Arity, typeCtorDecl.Attributes)
     {
       this.constructors = new List<DatatypeConstructor>();
+      this.selectors = new Dictionary<string, List<DatatypeSelector>>();
+    }
+
+    public void AddConstructor(DatatypeConstructor constructor)
+    {
+      this.constructors.Add(constructor);
+    }
+
+    public void AddSelector(DatatypeSelector selector)
+    {
+      if (!selectors.ContainsKey(selector.OriginalName))
+      {
+        selectors.Add(selector.OriginalName, new List<DatatypeSelector>());
```

Comment:
Is this where `selector.OriginalName` is something like `a` and `selector.Name` is something like `a#Pair`. With the new `->` syntax, I suggest supporting only `OriginalName`.

Reply:
Fixed now.